packages/langium/src/workspace/workspace-manager.ts

```diff
@@ -116,7 +116,8 @@ export class DefaultWorkspaceManager implements WorkspaceManager {
                 if (entry.isDirectory) {
                     await this.traverseFolder(workspaceFolder, entry.uri, fileExtensions, collector);
                 } else if (entry.isFile) {
-                    const document = this.langiumDocuments.getOrCreateDocument(entry.uri);
+                    const content = await this.fileSystemProvider.readFile(entry.uri);
```

Comment:
Here's a third place of `await this.fileSystemProvider.readFile(entry.uri)`.
I would prefer to encapsulate them within a single service, and `DocumentFactory` is the natural candidate (and it was like that before I think)